pkg/controllers/management/clusterprovisioner/provisioner.go

```diff
@@ -126,7 +130,11 @@ func (p *Provisioner) Remove(cluster *v3.Cluster) (runtime.Object, error) {
 }
 
 func (p *Provisioner) Updated(cluster *v3.Cluster) (runtime.Object, error) {
-	obj, err := v32.ClusterConditionUpdated.Do(cluster, func() (runtime.Object, error) {
+	if cluster.Spec.EKSConfig != nil {
```

Comment:
I think adding a debug in all of these skips to indicate Provisioner will not be operating on the cluster will help debugging later